zrml/global-disputes/src/lib.rs

```diff
@@ -40,26 +41,77 @@ mod pallet {
         sp_runtime::traits::StaticLookup,
         traits::{
             Currency, ExistenceRequirement, Get, IsType, LockIdentifier, LockableCurrency,
-            WithdrawReasons,
+            StorageVersion, WithdrawReasons,
         },
         Blake2_128Concat, BoundedVec, PalletId, Twox64Concat,
     };
     use frame_system::{ensure_signed, pallet_prelude::OriginFor};
     use sp_runtime::{
         traits::{AccountIdConversion, CheckedDiv, Saturating, Zero},
-        DispatchResult,
+        DispatchError, DispatchResult,
     };
     use sp_std::{vec, vec::Vec};
-    use zeitgeist_primitives::types::OutcomeReport;
+    use zeitgeist_primitives::{
+        traits::DisputeResolutionApi,
+        types::{Asset, Market, OutcomeReport},
+    };
     use zrml_market_commons::MarketCommonsPalletApi;
 
+    pub(crate) type MarketOf<T> = Market<
+        <T as frame_system::Config>::AccountId,
+        MarketCommonsBalanceOf<T>,
+        <T as frame_system::Config>::BlockNumber,
+        MomentOf<T>,
+        Asset<MarketIdOf<T>>,
+    >;
+    pub(crate) type BalanceOf<T> = <<T as Config>::Currency as Currency<AccountIdOf<T>>>::Balance;
+    pub(crate) type MomentOf<T> = <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Moment;
+    pub(crate) type MarketCommonsBalanceOf<T> =
+        <CurrencyOf<T> as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    pub(crate) type CurrencyOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Currency;
```

Comment:
You should be able to resolve that by using this typedef (let the compiler know that these types are supposed to be the same):

```rust
type MarketCommons: MarketCommonsPalletApi<
    AccountId = Self::AccountId,
    Currency = Self::Currency, // This line is new!
    BlockNumber = Self::BlockNumber,
>;
```

Reply:
54169e29cddb7b8511b2bd75adc8dd685681a2b2

Reply:
I'm a bit puzzled here. You add this line in 54169e2, but it appears as though you didn't even need to do that because you split this separation of two different balance/currency types over two files, one of which is only used for testing.

Reply:
No, doesn't hurt and definitely makes sense.